models/event.js

```diff
@@ -0,0 +1,116 @@
+'use strict';
+
+const Build = require('./build');
+const Joi = require('joi');
+const mutate = require('../lib/mutate');
+const Pipeline = require('./pipeline');
+const Scm = require('../core/scm');
+const Workflow = require('../config/workflow');
+
+const buildId = Joi.reach(Build.base, 'id');
+
+const MODEL = {
+    id: Joi
+        .string().hex().length(40)
+        .description('Identifier of this build')
+        .example('4b8d9b530d2e5e297b4f470d5b0a6e1310d29c5e'),
+    builds: Joi
+        .array().items(buildId)
+        .description('List of build IDs associated with this event')
+        .example(['8843d7f92416211de9ebb963ff4ce28125932878']),
+    commit: Scm.commit
+        .description('Commit related to the event'),
+    createTime: Joi
+        .string()
+        .isoDate()
+        .description('When this event was created')
+        .example('2038-01-19T03:14:08.131Z'),
+    creator: Scm.user
+        .description('Creator of the event'),
+    pipelineId: Joi.reach(Pipeline.base, 'id')
+        .description('The pipeline that the event belongs to')
+        .example('2d991790bab1ac8576097ca87f170df73410b55c'),
+    sha: Joi
+        .string().hex()
+        .description('SHA this project was built on')
+        .example('ccc49349d3cffbd12ea9e3d41521480b4aa5de5f'),
+    workflow: Workflow.workflow
+        .description('Workflow of the associated pipeline')
+        .example(['main', 'publish', 'deploy'])
+};
+
+module.exports = {
+    /**
+     * List of all fields in the model
+     * @property allKeys
+     * @type {Array}
+     */
+    allKeys: Object.keys(MODEL),
+
+    /**
+     * All the available properties of Event
+     *
+     * @property base
+     * @type {Joi}
+     */
+    base: Joi.object(MODEL).label('Event'),
+
+    /**
+     * Properties for Event that will be passed during a CREATE request
+     *
+     * @property create
+     * @type {Joi}
+     */
+    create: Joi.object(mutate(MODEL, [
+        'createTime',
+        'creator',
+        'pipelineId',
+        'sha'
+    ])).label('Create Event'),
```

Comment:
I feel like we won't need this and `update`. We aren't exposing these endpoints to users.